### PR TITLE
Update documentation for adding a dataset.

### PR DESCRIPTION
**Pull request by justinxzhao**

### CODE REVIEW

File: docs/developer_guide/add_a_dataset.md [hunk 1]
```diff
@@ -1,44 +1,127 @@
...
+
+# 2. Define a class that inherits from `ludwig.datasets.base_dataset.BaseDataset`
+
+We'll want to implement a class that inherits from `ludwig.datasets.base_dataset.BaseDataset` and implement the
+following methods:
 
 ```python
 @abc.abstractmethod
 def download_raw_dataset(self):
+    """Download the file from config.download_urls and save the file(s) as self.raw_dataset_path."""
```
Comment (dantreiman) [inline docs/developer_guide/add_a_dataset.md:29]: These doc strings are useful, and should be in the source code of BaseDataset as well if they aren't already.

File: docs/developer_guide/add_a_dataset.md [hunk 2]
```diff
@@ -1,44 +1,127 @@
...
     raise NotImplementedError()
 
-
 @abc.abstractmethod
 def process_downloaded_dataset(self):
+    """Process the dataset into a dataframe and save it as processed_dataset_path."""
```
Comment (dantreiman) [inline docs/developer_guide/add_a_dataset.md:34]:
```suggestion
    """Process the dataset into a dataframe and save it at self.processed_dataset_path."""
```

File: docs/developer_guide/add_a_dataset.md [hunk 3]
```diff
@@ -1,44 +1,127 @@
...
     raise NotImplementedError()
 
-
 @abc.abstractmethod
 def load_processed_dataset(self, split):
+    """Loads the processed data from processed_dataset_path into a Pandas DataFrame in memory.
```
Comment (dantreiman) [inline docs/developer_guide/add_a_dataset.md:39]:
```suggestion
    """Loads the processed data from self.processed_dataset_path into a Pandas DataFrame in memory.
```

File: docs/developer_guide/add_a_dataset.md [hunk 4]
```diff
@@ -1,44 +1,127 @@
...
+
+    :param split: Splits dataset along 'split' column if present.
```
Comment (dantreiman) [inline docs/developer_guide/add_a_dataset.md:41]: Clarify that the implementation of load_processed_dataset is responsible for splitting the data.

Docs should mention that `load_processed_dataset` returns a dataframe if no split, and a 3-tuple of train, val, test if split.

Also I think the split column should always have values 0: train, 1: validation, 2: test.

File: docs/developer_guide/add_a_dataset.md [hunk 5]
```diff
@@ -1,44 +1,127 @@
...
     raise NotImplementedError()
 ```
 
-For common steps (e.g., extracting zip files, downloading from Kaggle, etc.) a set of mixins are available for your subclass:
+For example:
+
+```python
+@register_dataset(name="wmt15")
+class WMT15(BaseDataset):
+    """French/English parallel texts for training translation models. Over 22.5 million sentences in French and
```
Comment (dantreiman) [inline docs/developer_guide/add_a_dataset.md:51]: Minor detail: line-break these docstrings inside code blocks so the reader doesn't have to scroll.

File: docs/developer_guide/add_a_dataset.md [hunk 6]
```diff
@@ -1,44 +1,127 @@
...
+class WMT15(CSVLoadMixin, IdentityProcessMixin, KaggleDownloadMixin, BaseDataset):
+    """French/English parallel texts for training translation models. Over 22.5 million sentences in French and
+    English.
+
+    Additional details:
+
+    https://www.kaggle.com/dhruvildave/en-fr-translation-dataset
+    """
+
+    def __init__(self, cache_dir=DEFAULT_CACHE_LOCATION, kaggle_username=None, kaggle_key=None):
```
Comment (dantreiman) [inline docs/developer_guide/add_a_dataset.md:79]: Break init into multiple lines so code block doesn't need to scroll horizontally.

**Review by dantreiman — APPROVED** [review]
Looks great!  Approved, with a few minor comments.

File: docs/developer_guide/add_a_dataset.md (hunk 1 above)
Comment (justinxzhao) [reply docs/developer_guide/add_a_dataset.md:29]: Agreed. I have a pending PR to add these.

File: docs/developer_guide/add_a_dataset.md (hunk 6 above)
Comment (justinxzhao) [reply docs/developer_guide/add_a_dataset.md:79]: Done.

**Pull request merged**